vm_supervisor/vm/firecracker/instance.py

```diff
@@ -115,5 +120,89 @@ async def wait_for_init(self) -> None:
 
     async def configure(self):
         """Configure the VM by sending configuration info to it's init"""
-        # TODO: Implement Cloud-init interface
         pass
+
+    async def _create_user_data_file(self) -> str:
+        """Creates user data configuration file for cloud-init tool"""
+
+        ssh_authorized_keys = self.resources.message_content.authorized_keys \
+            if self.resources.message_content.authorized_keys \
+            else []
+
+        config = {
+            "hostname": self.vm_hash,
+            "disable_root": False,
+            "ssh_pwauth": True,
```

Comment:
Password authentication should never be allowed outside of tests.

```suggestion
            "ssh_pwauth": False,
```